crates/edr_evm/src/chain_spec.rs

```diff
@@ -1,10 +1,24 @@
 use std::fmt::Debug;
 
-use edr_eth::transaction::SignedTransaction;
+use alloy_rlp::RlpEncodable;
+use edr_eth::{transaction::SignedTransaction, B256};
+use edr_rpc_eth::spec::{EthRpcSpec, RpcSpec};
 use revm::primitives::TxEnv;
+use serde::{de::DeserializeOwned, Serialize};
+
+use crate::{transaction::remote::EthRpcTransaction, EthRpcBlock, IntoRemoteBlock};
 
 /// A trait for defining a chain's associated types.
-pub trait ChainSpec {
+// Bug: https://github.com/rust-lang/rust-clippy/issues/12927
+#[allow(clippy::trait_duplication_in_bounds)]
+pub trait ChainSpec:
+    Debug
+    + alloy_rlp::Encodable
+    + RpcSpec<
+        RpcBlock<<Self as RpcSpec>::RpcTransaction>: EthRpcBlock + IntoRemoteBlock<Self>,
+        RpcTransaction: EthRpcTransaction,
+    > + RpcSpec<RpcBlock<B256>: EthRpcBlock>
+{
     /// The type of signed transactions used by this chain.
     type SignedTransaction: alloy_rlp::Encodable
```

Comment:
Just for my understanding: why were the bounds removed?

Reply:
I don't understand which part of the code you're referring to. Could you please elaborate?

The section where this comment refers to only adds trait bounds

Reply:
Sorry, put the comment on the wrong line: https://github.com/NomicFoundation/edr/pull/513/files/a60e79318138bf2cfeeb84801c823c5c0a625f45#r1642537614